devices/HumanDynamicsEstimator/HumanDynamicsEstimator.cpp

```diff
@@ -2292,51 +2293,39 @@ void HumanDynamicsEstimator::run()
         pImpl->berdyData.helper.extractLinkNetExternalWrenchesFromDynamicVariables(estimatedDynamicVariables,
                                                                                    pImpl->berdyData.estimates.linkNetExternalWrenchEstimates);
 
-//        pImpl->berdyData.estimates.linkNetExternalWrenchEstimates = pImpl->berdyData.estimates.task1_linkNetExternalWrenchEstimates;
-
-        // Using kindyn to compute link
-        iDynTree::KinDynComputations kinDynComputations;
-        kinDynComputations.loadRobotModel(pImpl->humanModel);
-        kinDynComputations.setFloatingBase(pImpl->humanModel.getLinkName(pImpl->berdyData.state.floatingBaseFrameIndex));
-        iDynTree::Vector4 quat;
-        quat.setVal(0, baseOrientation[0]);
-        quat.setVal(1, baseOrientation[1]);
-        quat.setVal(2, baseOrientation[2]);
-        quat.setVal(3, baseOrientation[3]);
-
-        iDynTree::Rotation rot;
-        rot.fromQuaternion(quat);
-
-        iDynTree::Position pos;
-        pos.setVal(0, basePosition[0]);
-        pos.setVal(1, basePosition[1]);
-        pos.setVal(2, basePosition[2]);
-
-        iDynTree::Transform baseTransform;
-        baseTransform.setPosition(pos);
-        baseTransform.setRotation(rot);
-
-        iDynTree::VectorDynSize s, sdot;
-        s.resize(jointsPosition.size());
-        sdot.resize(jointsPosition.size());
-        for (size_t idx = 0; idx < jointsPosition.size(); idx++)
-        {
-            s.setVal(idx, jointsPosition[idx]);
-            sdot.setVal(idx, jointsVelocity[idx]);
-        }
+        // Express the dummy sources wrench esitmates with orientation of world frame
+        if (pImpl->expressDummyWrenchEstimatesInWorldOrientation) {
+
+            for (auto& element : pImpl->wrenchSourceNameAndType) {
+
+                // Iterate over all the dummy wrench sources
+                if (element.second == WrenchSourceType::Dummy) {
+
+                    // Get link index
+                    std::string linkName = element.first;
+                    int linkIndex = pImpl->humanModel.getLinkIndex(element.first);
+
+                    // Get link to world transform
+                    iDynTree::Transform world_H_link = kinDynComputations.getWorldTransform(linkName);
+
+                    // Set the position to zero
+                    world_H_link.setPosition(iDynTree::Position(0.0, 0.0, 0.0));
+
+                    // Get extracted wrench
+                    iDynTree::Wrench linkWrench = pImpl->berdyData.estimates.linkNetExternalWrenchEstimates(linkIndex);
+
+                    // Transform extracted wrench estimate with orietation of world frame
+                    Eigen::Matrix<double,6,1> transformedWrenchEigen = iDynTree::toEigen(world_H_link.asAdjointTransformWrench()) * iDynTree::toEigen(linkWrench.asVector());
```

Comment:
I think we don't need to explicitly define an `Eigen::Matrix` here, but it could have been done as follow
```
iDynTree::Wrench transformedLinkWrench;
iDynTree::toEigen(transformedLinkWrench) = iDynTree::toEigen(world_H_link.asAdjointTransformWrench()) * iDynTree::toEigen(linkWrench.asVector());
```

Reply:
There seems to be no operator overload for `=` to perform such an assignment.